client/internal/statemanager/manager.go

```diff
@@ -179,14 +179,18 @@ func (m *Manager) PersistState(ctx context.Context) error {
 		return nil
 	}
 
+	bs, err := json.MarshalIndent(m.states, "", "    ")
+	if err != nil {
+		return fmt.Errorf("marshal states: %w", err)
+	}
+
```

Comment:
Why do not unlock the m.mu? It blocks the Updated function calls in the worst case for 5 sec. After this point, the code does not touch the m.state so make no sense to protect it.

Reply:
We still need to clean up the `dirty` map further down. If we unlock here we might clear `dirty` with new entries that haven't been yet written.

Reply:
ok

Reply:
The log running code (5 sec) really does not cause any issue outside of the statemanager?

Reply:
It can delay updating iptables/nftables/routes/dns on network map updates. Because of routes it might also delay p2p connections